pkg/standalone-utils/test/EulerWrapping.test.ts

```diff
@@ -0,0 +1,660 @@
+          ).wait();
+
+          const relayerIsSender = TypesConverter.toAddress(tokenSender) === relayer.address;
+          if (!relayerIsSender) {
+            expectTransferEvent(
+              receipt,
+              {
+                from: tokenSender.address,
+                to: relayer.address,
+                value: amount,
+              },
+              DAI
+            );
+          }
+          expectTransferEvent(
+            receipt,
+            {
+              from: relayer.address,
+              to: mockEulerProtocol.address,
+              value: amount,
+            },
+            DAI
+          );
+
+          const relayerIsRecipient = TypesConverter.toAddress(tokenRecipient) === relayer.address;
+          expectTransferEvent(
+            receipt,
+            {
+              from: ZERO_ADDRESS,
+              to: relayer.address,
+              value: expectedulerAmount,
+            },
+            eDAI
+          );
+          if (!relayerIsRecipient) {
+            expectTransferEvent(
+              receipt,
+              {
+                from: relayer.address,
+                to: tokenRecipient.address,
+                value: expectedulerAmount,
+              },
+              eDAI
+            );
+          }
+        });
+      }
+    });
+
+    describe('unwrap Euler', async () => {
+      let tokenSender: Account, tokenRecipient: Account;
+
+      beforeEach(async () => {
+        // Euler protocol does not have any underlying tokens
+        // so we wint them.
+        DAI.mint(mockEulerProtocol.address, fp(5000));
+      });
+
+      context('sender = senderUser, recipient = relayer', () => {
+        beforeEach(async () => {
+          await eDAI.connect(senderUser).approve(vault.address, fp(100));
+          tokenSender = senderUser;
+          tokenRecipient = relayer;
+        });
+        testUnwrap();
+      });
+
+      function testUnwrap(): void {
+        it('unwraps with immediate amounts', async () => {
+          const receipt = await (
+            await relayer
+              .connect(senderUser)
+              .multicall([encodeUnwrap(eDAI.address, tokenSender, tokenRecipient, amount)])
+          ).wait();
+
+          const unwrappedAmount = await eDAI.convertBalanceToUnderlying(amount);
+
+          const relayerIsSender = TypesConverter.toAddress(tokenSender) === relayer.address;
+          if (!relayerIsSender) {
+            expectTransferEvent(
+              receipt,
+              {
+                from: TypesConverter.toAddress(tokenSender),
+                to: TypesConverter.toAddress(relayer),
+                value: amount,
+              },
+              eDAI
+            );
+          }
+          expectTransferEvent(
+            receipt,
+            {
+              from: mockEulerProtocol.address,
+              to: TypesConverter.toAddress(relayer),
+              value: await eDAI.convertBalanceToUnderlying(amount),
```

Comment:
Let's use `unwrappedAmount` here since we already computed it above.
```suggestion
              value: unwrappedAmount,
```